src/Files.App/Services/Settings/GeneralSettingsService.cs

```diff
@@ -315,6 +315,30 @@ public ShellPaneArrangement ShellPaneArrangementOption
 			set => Set((long)value);
 		}
 
+		public int MainWindowWidth
+		{
+			get => Get(int.MaxValue);
+			set => Set(value);
+		}
+
+		public int MainWindowHeight
+		{
+			get => Get(int.MaxValue);
+			set => Set(value);
+		}
+
+		public int MainWindowPositionX
+		{
+			get => Get(int.MaxValue);
+			set => Set(value);
+		}
+
+		public int MainWindowPositionY
+		{
+			get => Get(int.MaxValue);
+			set => Set(value);
+		}
```

Comment:
Can we use a tuple to condense the code?

Reply:
I'm gonna use ApplicationData as you suggested.
The key names would be those.

- MainWindowWidth
- MainWindowHeight
- MainWindowPositionX
- MainWindowPositionY

What do you think?

Reply:
Also, one more last thing we should concern is when monitor info has changed since last time the app launched.
If monitor resolution changed WinUIEx doesn't restore info and reset info.

Reply:
The key names aren't that important but we might as well use the same ones as WinUIEx for the least disruption.

Reply:
WinUIEx creates a key named $"WindowPersistance_{PersistenceId}" and puts value that is converted into base64 string of WINDOWPLACEMENT struct data.

Reference:

https://github.com/dotMorten/WinUIEx/blob/ea41e17bc3ecba48491db9920a9ed7348f62d44b/src/WinUIEx/WindowManager.cs#L428-L469

Reply:
I think we don't want such a complexity so a set of keys is preferable

Reply:
We would have to migrate the existing settings either way.

Reply:
You don't want to drop support of the window size before this removal? Even just one time?

Btw if so, when are we gonna such migration code?

Reply:
I meant that the existing window size settings shouldn't get lost.

Reply:
But it requires a migration code.
When are we gonna remove it(shouldn't exist forever right?)?

Reply:
A couple months is good, but if migrating is complicated, we can just start fresh.

Reply:
It's not that complicated, I'll make a code of it and then please take a look at later:)